android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java

```diff
@@ -799,6 +801,20 @@ public boolean shouldOverrideUrlLoading(WebView view, WebResourceRequest request
       return this.shouldOverrideUrlLoading(view, url);
     }
 
+    @TargetApi(Build.VERSION_CODES.LOLLIPOP)
+    @Override
+    public WebResourceResponse shouldInterceptRequest(WebView view, WebResourceRequest request) {
+      String cookie = request.getRequestHeaders() == null ? null : request.getRequestHeaders().get("cookie");
```

Comment:
Hi, just wondering if "cookie" should be case insensitive?  It didn't work when I declared my headers as `headers={{ Cookie: "key=value" }}`.  Appears to be case insensitive on iOS.

Reply:
Hi @dwxw, Did you check this in Android? Did you check `headers={{ cookie: "key=value" }}`?

Reply:
Yes I tried your change and compared it to the existing iOS behaviour.  The headers prop works with `Cookie` or `cookie` in iOS, but with this change only `cookie` works on Android.  [AFAIK](https://stackoverflow.com/questions/5258977/are-http-headers-case-sensitive/5259004#5259004) the http header should be case-insensitive so you should handle both cases.

Reply:
@1rahul1 Do you agree with @dwxw? If you want to make this case insensitive, I'll merge & release after. Or if not, I can merge and release now.

Reply:
@dwxw, thanks for checking that. I checked it on Android 9.0 but wasn't able to reproduce the issue which you pointed out. For me, it worked for both cases. Can you give more details on which version of Android used and how to reproduce it?

Reply:
@jamonholmgren I checked this and it worked for me for both the cases. In the headers from [ `request.getRequestHeaders()`](https://github.com/1rahul1/react-native-webview/blob/9970504f2a6601531cc8557bd82f4342f9c1b3f2/android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java#L831), the **keys** of the headers are in **lowercase** for the headers passed in the source prop, **irrespective** of the case of the key passed. [Here](https://github.com/1rahul1/react-native-webview/blob/9970504f2a6601531cc8557bd82f4342f9c1b3f2/android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java#L831), I have used "cookie" which is in lowercase. So, I think the case-insensitive case is already handled.

Reply:
I've had another look and I still have this problem where only "cookie" works for me. The server does not receive the cookie unless I use "cookie", but after digging into this code it must be something else.  I've looked at `request.getRequestHeaders()` myself and it always returns a lowercase "cookie" no matter the case I use in the `headers` prop.  So not sure what my problem is but I think @1rahul1's code looks good.  Sorry for any confusion.